### ISSUE TITLE
Shift back to a more restrictive layout system

### ISSUE THREAD

**Issue by ang-zeyu** [MarkBind/markbind#2077]
### Please confirm that you have searched existing issues in the repo

Yes, I have searched the existing issues

### Any related issues?

Primarily #1961

### What is the area that this feature belongs to?

Author Usability

### Is your feature request related to a problem? Please describe.

Several issues have been found regarding the new layout system introduced in V3.
1. when authors attempt to upgrade their MarkBind version https://github.com/MarkBind/markbind/pull/1403, it is difficult to migrate layout related fixes/changes to user (author) files.
2. features such as sticky headers that rely on assumptions of the layout structure are more unwieldy for authors to use, since they are in control of the layout file. It requires moderate understanding of css internals which may go against product goals. There may be more such features to come as well.

This comment documents briefly how the old system looked like:
https://github.com/MarkBind/markbind/issues/1961#issuecomment-1181832361

Original motivations for the migration:
1. A huge part of the motivation for the migration in the first place was reducing the complexity (duplication). Back then, `<include>` usage was not supported in layout files, so layout duplication is rampant across the layout folders and individual layout files.
2. To reduce the number of layout files and make author maintainence of layouts simpler and more flexible with a one file system. (to include common components, one simply had to `<include/>` them)

### Describe the solution you'd like

This ticket is for proposing ideas on what an ideal layout system for MarkBind should look like considering all the above concerns.

Maintaining `<include/>` compatibility should be a must in any case, as it is a very core feature.

**Example ideas:**

Complete migration back to the old system is not necessary, we can find ways to work with the current situation.

For example, one simple idea would be to require layout files to have some "sections"
```
---
type: header
... markup ...
---
type: sitenav (corresponding to id="site-nav" in the docs)
... markup ...
---
type: pagenav (corresponding to id="page-nav")
... markup ...
---
type: content(corresponding to id="content-wrapper")
... markup ...
---
...
```

On the other hand we shouldn't shy away from going back to a separate file system as well. Some of the earlier issues we had would be dealt with by allowing use of `<include/>`s.

One way of improving the number of files would be to have the `headers`, `footers`, `navigation`, etc. directories in this [comment](https://github.com/MarkBind/markbind/issues/1961#issuecomment-1181832361) but extract the `layout1/2/3/etc.` layout folders to JSON configuration in the site config file or elsewhere, pointing to files inside `headers`, `footers`, `navigation`.

These are just some simple ideas.

**Comment by ang-zeyu** [comment]
@MarkBind/active-devs, specifically anyone taking 3282, I think this would be a good ticket for some collaborative discussion and product thinking. You can come up with ideas, discuss amongst yourselves here or offline (but please document your contributions somewhere), then file a follow up ticket to implement the best one.

cc @jonahtanjz you could also help influence the final decision.

Feel free to ping me or @damithc if you require more context, understand it may be somewhat confusing as it is very old stuff.

**Comment by ang-zeyu** [comment]
This should be the highest priority thing currently as well.

**Comment by raysonkoh** [comment]
Just to clarify, am I correct in understanding that the goal of this issue is to ultimately limit the amount of customization that a user can make to the layout (i.e. we allow user to modify key components of the layout such as header, pagenav, sitenav, etc, but not so much as to how these components are tied together)? And this is so as to allow for easier layout-related fixes and less complexity on the user side?

**Comment by tlylt** [comment]
Agree that the current layout system can be improved.

I think it would be great if there are any more details on the pros and cons of the previous layout system (from a user's POV). It could be the case that the new layout system appears to be causing more problems simply because there's more usage, more migration, etc. And that the benefits could be overlooked as they tend to be less mentioned when we look at the number of issues found around it. We should have an extensive comparison before embarking on this.

**Comment by ang-zeyu** [comment]
> limit the amount of customization

Yes. Fwiw, other than a handful of pages (e.g. marketing pages), my rough observations has also been that webpages always do follow a fixed structure containing certain core components most of the time (e.g. header, sidebar, footer, main area). So customization may (discuss and agree on this) not be as valuable as we thought. Likely, you can still find ways to make complex pages by omitting headers, sidebars, etc. and writing everything in a "main area".

> We should have an extensive comparison before embarking on this.

Please do. Shifting back completely is not a must, you can try to find ways to work with the current system to solve the mentioned problems. But on the other hand, don't shy away from completely shifting the system as well.

Another thing you can do to preview the old system is check out the old docs roughly around this commit. https://github.com/MarkBind/markbind.github.io/commit/5fe1d592546583d1dd3529ac823605c6871e0e75. (use `http-server` or similar to serve locally)

**Comment by jovyntls** [comment]
One way could be to provide a small number of fixed choices (templates) for layouts.
